### PR TITLE
LPD-26851 - Feat. Prevent self-referenced branches in the treeview

### PR DESCRIPTION
**Pull request by juanjofgliferay**
## Task
https://liferay.atlassian.net/browse/LPD-26851

## Issue
`StructuredContent` schema has self reference objects that causes an infinite loop, preventing the tree from loading in the DSM Visualization Modes pages.

## Solution
Prevent loading children if they are already part of the parent tree. 

This task fixes DataSetFragment#CanUseStructuredContentSchema test

## UI
[Screencast from 2024-05-28 14-39-42.webm](https://github.com/liferay-frontend/liferay-portal/assets/132653342/b91fcfae-f166-4dd9-af55-d0e16de992da)

### CODE REVIEW

**Comment by juanjofgliferay** [comment]
image
[Testray report](https://testray.liferay.com/home/-/testray/case_results/2572721155?redirect=https%3A%2F%2Ftestray.liferay.com%2Fhome%2F-%2Ftestray%2Fcase_results%2Findex%3FtestrayBuildId%3D2572720737%26cur%3D1%26delta%3D200%26orderByCol%3Dstatus_sortable%26orderByType%3Dasc)

**Review by markocikos — COMMENTED** [review]
@juanjofgliferay A minor improvement suggestion inline.

In general, solution is simple and good enough for now. We may want to improve it down the line, by implementing dynamic loading of children, that happens on expand.

File: modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/utils/getFields.ts [hunk 1]
```diff
@@ -65,20 +65,30 @@ function getValidFields({
...
+								''
+							),
+							schemas,
+						}),
+						label: propertyKey,
+						name: `${contextPath}${propertyKey}${FDS_ARRAY_FIELD_NAME_PARENT_SUFFIX}`,
+						sortable: false,
+						type: type ? type : 'array',
+					});
+				}
```
[the remark this answers is not in the document]
Comment (juanjofgliferay) [reply modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/utils/getFields.ts:91]: Sure

**Comment by juanjofgliferay** [comment]
ci:test:sf

**Review by markocikos — APPROVED** [review]
@juanjofgliferay LGTM! :sparkles:

**Pull request closed without merging**